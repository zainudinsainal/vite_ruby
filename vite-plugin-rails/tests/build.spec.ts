import { resolve } from 'path'
import { readFileSync } from 'fs'
import { beforeAll, describe, test, expect } from 'vitest'
import execa from 'execa'
import { glob } from 'tinyglobby'

const projectRoot = resolve(__dirname, '../')
const exampleDir = `${projectRoot}/example`

describe('config', () => {
  beforeAll(async () => {
    await execa('npm', ['run', 'build'], { stdio: process.env.DEBUG ? 'inherit' : undefined, cwd: exampleDir })
  }, 60000)

  test('generated files', async () => {
    const outDir = `${exampleDir}/public/vite`
    const files = await glob('**/*', { cwd: outDir, onlyFiles: true })

    // Check for expected file patterns instead of specific hashes
    const hasFilePattern = (pattern: RegExp) => files.some(file => pattern.test(file))

    // CSS files
    expect(hasFilePattern(/^assets\/app-[a-zA-Z0-9_-]+\.css$/)).toBe(true)
    expect(hasFilePattern(/^assets\/app-[a-zA-Z0-9_-]+\.css\.br$/)).toBe(true)
    expect(hasFilePattern(/^assets\/app-[a-zA-Z0-9_-]+\.css\.gz$/)).toBe(true)
    expect(hasFilePattern(/^assets\/sassy-[a-zA-Z0-9_-]+\.css$/)).toBe(true)
    expect(hasFilePattern(/^assets\/theme-[a-zA-Z0-9_-]+\.css$/)).toBe(true)
    expect(hasFilePattern(/^assets\/vue-[a-zA-Z0-9_-]+\.css$/)).toBe(true)

    // JS files
    expect(hasFilePattern(/^assets\/external-[a-zA-Z0-9_-]+\.js$/)).toBe(true)
    expect(hasFilePattern(/^assets\/external-[a-zA-Z0-9_-]+\.js\.map$/)).toBe(true)
    expect(hasFilePattern(/^assets\/index-[a-zA-Z0-9_-]+\.js$/)).toBe(true)
    expect(hasFilePattern(/^assets\/main-[a-zA-Z0-9_-]+\.js$/)).toBe(true)
    expect(hasFilePattern(/^assets\/vue-[a-zA-Z0-9_-]+\.js$/)).toBe(true)

    // Images
    expect(hasFilePattern(/^assets\/logo-[a-zA-Z0-9_-]+\.svg$/)).toBe(true)
    expect(hasFilePattern(/^assets\/logo-[a-zA-Z0-9_-]+\.png$/)).toBe(true)

    // Static files
    expect(files).toContain('index.html')
    expect(files).toContain('index.html.br')
    expect(files).toContain('index.html.gz')

    const parseManifest = (path: string) => JSON.parse(readFileSync(`${outDir}/.vite/${path}`, 'utf-8'))
    const manifest = parseManifest('manifest.json')
    const manifestAssets = parseManifest('manifest-assets.json')

    expect({ ...manifest, ...manifestAssets }).toMatchSnapshot()
  })
})
