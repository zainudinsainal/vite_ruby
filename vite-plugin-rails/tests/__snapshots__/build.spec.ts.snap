// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config > generated files 1`] = `
{
  "../assets/external.js": {
    "file": "assets/external-BUA3rgrF.js",
    "integrity": "sha384-aarzsPiJ+RIlIMD390CUw1EWAKENm7zypFnKXvMt8kolSFmRbXHAI1Z3bOKxe5WL",
    "isEntry": true,
    "name": "app/assets/external.js",
    "src": "../assets/external.js",
  },
  "../assets/logo.png": {
    "file": "assets/logo-CPmPqqKk.png",
    "integrity": "sha384-IGw3tfo4Xt0lp6sO9t5mYtILwfBxDFR8sRCtroxGjhE3r9UxNYgtimkCNzsJU9AP",
    "src": "app/assets/logo.png",
  },
  "../assets/theme.css": {
    "file": "assets/theme-CK6FhHUh.css",
    "integrity": "sha384-tXSaWp9h7f9IDyqcyqxE2oHEre7hVTMpGM1IZPUdwc9do71+G5Fb37fW6KleExPw",
    "isEntry": true,
    "src": "../assets/theme.css",
  },
  "entrypoints/app.css": {
    "file": "assets/app-BhUNvwbY.css",
    "integrity": "sha384-j2SX6yHUu2YQv/067X7HfP4nv2AUV8u5F+we/wI1eQKP0FweNgIiRxCg6vhW6yJE",
    "isEntry": true,
    "src": "entrypoints/app.css",
  },
  "entrypoints/frameworks/vue.js": {
    "css": [
      "assets/vue-BqFVa9O9.css",
    ],
    "file": "assets/vue-AiHcjmSE.js",
    "integrity": "sha384-P6xS4AtvWWb80KwaNsFL32BdBXCJHCIvj3DCLssi3jUcWTwVIT3pVkRghSMsF7j2",
    "isEntry": true,
    "name": "entrypoints/frameworks/vue.js",
    "src": "entrypoints/frameworks/vue.js",
  },
  "entrypoints/main.ts": {
    "css": [
      "assets/app-BhUNvwbY.css",
      "assets/theme-CK6FhHUh.css",
    ],
    "file": "assets/main-DEjkSHLE.js",
    "imports": [
      "entrypoints/frameworks/vue.js",
    ],
    "integrity": "sha384-K2NguofOgA2PX5gQqPIA1lQmO5szUiqTIBUrnwBd07LOb3pj8GlgjXsZ6bhB5sIm",
    "isEntry": true,
    "name": "entrypoints/main.ts",
    "src": "entrypoints/main.ts",
  },
  "entrypoints/sassy.scss": {
    "file": "assets/sassy-D5kz_As0.css",
    "integrity": "sha384-4tkn835gOI7C29uZzlN/PGxd+XyQTG1/9AHpbWYbYLbgWZ1UmDDBFW1iq43MC70L",
    "isEntry": true,
    "src": "entrypoints/sassy.scss",
  },
  "images/logo.png": {
    "file": "assets/logo-Bh_XL1Xl.png",
    "integrity": "sha384-zHCg6OQ1T1xX21KMsW9Bb2ECIIMeec41+3KtT0gygdsrmm/Z5qD5TVtTPhyJNGot",
    "src": "images/logo.png",
  },
  "images/logo.svg": {
    "file": "assets/logo-Nuo4H8F_.svg",
    "integrity": "sha384-qQuFPGf9f7DHitjACIjHvY2GwlcFjg+HatTRP3bwG7iAX8MzOW4Sr2dfca6uZlDA",
    "src": "images/logo.svg",
  },
  "index.html": {
    "assets": [
      "assets/logo-Bh_XL1Xl.png",
    ],
    "css": [
      "assets/sassy-D5kz_As0.css",
      "assets/app-BhUNvwbY.css",
      "assets/theme-CK6FhHUh.css",
    ],
    "file": "assets/index-U2H-XnNn.js",
    "imports": [
      "entrypoints/main.ts",
      "entrypoints/frameworks/vue.js",
    ],
    "integrity": "sha384-DjqU+kDxDLID8DTt6F6hKnCqUaQIjaauOspgzY6a3ZARVCKDSktMt25vK3FGVVsj",
    "isEntry": true,
    "name": "index.html",
    "src": "index.html",
  },
  "logo.png": {
    "file": "assets/logo-CPmPqqKk.png",
    "integrity": "sha384-IGw3tfo4Xt0lp6sO9t5mYtILwfBxDFR8sRCtroxGjhE3r9UxNYgtimkCNzsJU9AP",
    "src": "logo.png",
  },
  "logo.svg": {
    "file": "assets/logo-Nuo4H8F_.svg",
    "integrity": "sha384-qQuFPGf9f7DHitjACIjHvY2GwlcFjg+HatTRP3bwG7iAX8MzOW4Sr2dfca6uZlDA",
    "src": "logo.svg",
  },
}
`;
